### PR TITLE
Dequeue for reordering messages.

### PR DESCRIPTION
**Pull request by defectiveAi**

### CODE REVIEW

File: source/Energinet.DataHub.PostOffice.Domain/Services/MarketOperatorDataDomainService.cs [hunk 1]
```diff
@@ -94,6 +94,20 @@ await _dequeueCleanUpSchedulingService
                 .ConfigureAwait(false);
         }
 
+        public async Task Acknowledge2Async(Bundle bundle)
+        {
+            if (bundle == null)
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.PostOffice.Domain/Services/MarketOperatorDataDomainService.cs:99]: Is Guard a joke to you?

File: source/Energinet.DataHub.PostOffice.Infrastructure/Common/CosmosExtensions.cs [hunk 2]
```diff
@@ -66,6 +66,26 @@ public static async Task<T> SingleAsync<T>(this IAsyncEnumerable<T> enumerable)
...
+            }
+
+            var result = enumerator.Current;
+
+            if (await enumerator.MoveNextAsync())
+            {
+                throw new InvalidOperationException("The collection has more than one element.");
+            }
+
+            return result;
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.PostOffice.Infrastructure/Common/CosmosExtensions.cs:86]: Couldn't you just return `enumerator.Current`. Save that variable

File: source/Energinet.DataHub.PostOffice.Infrastructure/Repositories/DataAvailableNotificationRepository.cs [hunk 3]
```diff
@@ -228,6 +234,42 @@ await foreach (var document in query.AsCosmosIteratorAsync().ConfigureAwait(fals
             }
         }
 
+        public async Task AcknowledgeAsync(Bundle bundle)
+        {
+            if (bundle == null)
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.PostOffice.Infrastructure/Repositories/DataAvailableNotificationRepository.cs:239]: Guard por favor

File: source/Energinet.DataHub.PostOffice.Infrastructure/Common/CosmosExtensions.cs (hunk 2 above)
Comment (defectiveAi) [reply source/Energinet.DataHub.PostOffice.Infrastructure/Common/CosmosExtensions.cs:86]: No, it cannot be saved.

File: source/Energinet.DataHub.PostOffice.Domain/Services/MarketOperatorDataDomainService.cs (hunk 1 above)
Comment (defectiveAi) [reply source/Energinet.DataHub.PostOffice.Domain/Services/MarketOperatorDataDomainService.cs]: Fixed

File: source/Energinet.DataHub.PostOffice.Infrastructure/Repositories/DataAvailableNotificationRepository.cs (hunk 3 above)
Comment (defectiveAi) [reply source/Energinet.DataHub.PostOffice.Infrastructure/Repositories/DataAvailableNotificationRepository.cs]: Fixed

File: source/Energinet.DataHub.PostOffice.Domain/Services/IMarketOperatorDataDomainService.cs [hunk 4]
```diff
@@ -70,7 +70,13 @@ public interface IMarketOperatorDataDomainService
         /// <summary>
         /// Acknowledges the current bundle, as returned by GetNextUnacknowledgedAsync.
         /// </summary>
-        /// <param name="bundle">The the bundle that is being acknowledged.</param>
+        /// <param name="bundle">The bundle that is being acknowledged.</param>
         Task AcknowledgeAsync(Bundle bundle);
+
+        /// <summary>
+        /// Acknowledges the current bundle, as returned by GetNextUnacknowledgedAsync.
```
Comment (michaelmajgaard) [inline source/Energinet.DataHub.PostOffice.Domain/Services/IMarketOperatorDataDomainService.cs:77]: Is it possible to reference Methods in XML docs?

**Review by michaelmajgaard — APPROVED** [review]

**Review by michaelmajgaard — APPROVED** [review]

**Review by mbusk88 — COMMENTED** [review]
Nice name convention :) I'll use that!

**Pull request merged**